### ISSUE TITLE
Analysis for choosing the nominal pose

### ISSUE THREAD

**Issue by vvasco** [cht-iit/design-setup-bp#10]
I report here the results of the analysis done to choose **nominal target pose**.

As stated in #5, we want to select a  **nominal target pose** such that:
- the **error in position** is _low_: I computed this as Euclidean distance between desired and reached the target and set a threshold of `1cm`;
- the **error in orientation** is _low_: I computed this as $\sin \theta$, where $\theta$ is the angle between desired and reached orientation (the closer to 0, the closer desired and reached orientation); for this, I set a threshold of `0.2` (corresponding to $\theta \approx 10 \thinspace \text{deg}$);
- the **joint configuration** has to stay within the joint limits: I computed the distance between each joint (of left arm and torso) and their lower and upper limits and set a threshold of `10.0 degrees`.

In order to perform the analysis, I selected a bunch of targets to reach and evaluated the aforementioned errors.
The following table summarizes the results for different targets (averaged over 3 repetitions).

|  Target                            |  Err <br> pos <br> [cm] |  Err <br> ori   | Shoulder  <br> [deg]           |   Elbow <br> [deg] | Wrist <br> [deg]  |  Torso <br> [deg] |  
|  ------                            |  -------------|  -------   | ----------              |   -------- |  ------- | -------- |  
|  Target 1: <br> xd = -0.6 0.0 -0.1 <br> ori = 0.0 0.0 1.0 3.14   |       **9**    |     **0.8**    |  pitch = [**0.1** 105.1] <br> roll = [55.2 104.9] <br> yaw = [51.7 65.3]   | [27.2 63.8] |  prosup = [78.0 42.0] <br> pitch = [58.8 46.2] <br> yaw = [25.4 19.6] | yaw = [99.3 **0.7**] <br> roll = [29.9 30.1] <br> pitch = [70.0 20.1] <br>         |
|  Target 2: <br>xd = -0.5 0.0 -0.1 <br> ori = 0.0 0.0 1.0 3.14   |      **2.3**    |     0.057    |  pitch = [**0.3** 104.8] <br> roll = [64.2 95.8] <br> yaw = [93.3 23.7]   | [18.0 73.0] |  prosup = [119.7 **0.3**] <br> pitch = [65.5 39.5] <br> yaw = [38.9 **6.1**] | yaw = [99.1 **0.9**] <br> roll = [29.8 30.1] <br> pitch = [64.7 25.2] <br>         |
|  Target 3: <br>xd = -0.4 0.0 -0.1 <br> ori = 0.0 0.0 1.0 3.14   |       **2.1**    |     0.054    |  pitch = [28.1 77.0] <br> roll = [28.2 131.7] <br> yaw = [85.0 32.0]   | [**2.0** 89.0] |  prosup = [120.0 **0.03**] <br> pitch = [46.8 58.2] <br> yaw = [45.0 10.1] | yaw = [75.1 25.0] <br> roll = [29.8 30.2] <br> pitch = [50.6 39.3] <br>         |  
|  Target 4: <br>xd = -0.4 0.0 0.0 <br> ori = 0.0 0.0 1.0 3.14    |       **1.3**    |     0.047    |  pitch = [31.3 73.7] <br> roll = [25.9 134.1] <br> yaw = [82.7 34.2]   | [10.3 80.7] |  prosup = [116.6 3.4] <br> pitch = [64.7 40.2] <br> yaw = [45.0 **0.02**] | yaw = [72.6 27.4] <br> roll = [29.8 30.2] <br> pitch = [34.2 55.9] <br>         |  
|  Target 5: <br>xd = -0.3 0.0 -0.1 <br> ori = 0.0 0.0 1.0 3.14   |       0.78    |     0.16    |  pitch = [61.9 43.1] <br> roll = [24.5 135.5] <br> yaw = [112.7 **4.2**]   | [**3.7** 87.3] |  prosup = [71.8 48.2] <br> pitch = [28.9 76.1] <br> yaw = [45.0 **0.02**] | yaw = [83.2 16.8] <br> roll = [29.8 30.1] <br> pitch = [25.3 64.7] <br>         |  
|  Target 6: <br>xd = -0.35 0.0 -0.1 <br> ori = 0.0 0.0 1.0 3.14   |       **1.2**    |     0.10    |  pitch = [50.0 55.0] <br> roll = [24.8 135.2] <br> yaw = [79.4 37.6]   | [**3.3** 87.6] |  prosup = [113.9 **6.1**] <br> pitch = [40.3 64.7] <br> yaw = [45.0 **0.02**] | yaw = [79.4 20.7] <br> roll = [29.8 30.2] <br> pitch = [37.4 52.6] <br>         |  
|  Target 7: <br>xd = -0.35 0.0 0.0 <br> ori = 0.0 0.0 1.0 3.14   |       0.77    |     0.12    |  pitch = [49.7 55.2] <br> roll = [25.3 134.7] <br> yaw = [89.8 27.2]   | [15.4 75.6] |  prosup = [100.7 19.3] <br> pitch = [60.1 44.8] <br> yaw = [45.0 **0.01**] | yaw = [73.7 26.3] <br> roll = [29.8 30.2] <br> pitch = [20.0 70.0] <br>         |
|  Target 8: <br>xd = -0.2 0.0 -0.1 <br> ori = 0.0 0.0 1.0 3.14    |       0.57    |     **0.51**    |  pitch = [78.9 26.1] <br> roll = [23.2 136.7] <br> yaw = [116.6 **0.38**]   | [38.9 52.1] |  prosup = [84.1 35.9] <br> pitch = [30.1 74.9] <br> yaw = [45.0 **0.04**] | yaw = [67.3 32.7] <br> roll = [29.9 30.1] <br> pitch = [20.1 70.0] <br>         |  

I highlighted results exceeding the thresholds (and thus bad targets):
- errors in position higher than `1 cm`;
- errors in orientation higher than `0.2`;
- joints distances from their upper and lower limits lower than `10 degrees`.

Overall, **targets 5** and **7** are the only ones reached with both errors below the thresholds, but **target 7** is reached with a joint configuration that fits better the limits.
Regarding the other targets, we can notice that:

- _target 1_ is not reached at all, as both errors in position and orientation are higher than the thresholds. Furthermore, the shoulder pitch and the torso yaw reach the lower and the upper limit respectively. This is an indication that the target is too far from the robot;
- _target 2_ is still far from the robot (high error on position and shoulder pitch and the torso yaw close to the limits);
- _target 3, 4 and 6_ are reached with a reasonable joint configuration but the error on position is over the threshold.
- _target 8_ has a high error on the orientation. Furthermore, the shoulder yaw reaches the upper limit. These are both indications that the target is too close to the robot.

So far, **target 7** seems the most reasonable candidate as nominal pose.
Finally I report here the errors over time we have for target 7:

| Err pos | Err ori |
|--- |--- |
|<img src=https://user-images.githubusercontent.com/9716288/88427679-b7a4cf80-cdf3-11ea-8c18-e3cec16010ee.jpg width="350"> | <img src=https://user-images.githubusercontent.com/9716288/88427709-c4292800-cdf3-11ea-9748-dce4f5dc2d4a.jpg width="350"> |
| _Legend_: <br> _blue: error_ <br> _magenta: threshold_  |

| Joint configuration |
| --- |
| <img src=https://user-images.githubusercontent.com/9716288/88427811-f5a1f380-cdf3-11ea-942a-4bd2e066fbdd.jpg width="650"> |
| _Legend_: <br> _black: dist from lower limit_ <br> _red: dist from upper limit_ <br> _magenta: threshold_  |

We can see that the position and errors go down (when the reaching action starts) and stay below the thresholds, apart from the peak in the position error.
Basically the hand goes back for `~0.5s` and then back again to the target. I noticed this quite often also with other targets.
The joint configuration also looks reasonable, as the distances stay above the threshold (apart from the wrist yaw, but this is because we are assuming that the object is in the center and thus `y=0`).

cc @Nicogene @pattacini @vtikha @maggia80
